tests/native_token_bridge.go

```diff
@@ -0,0 +1,180 @@
+package tests
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	nativetokendestination "github.com/ava-labs/teleporter/abi-bindings/CrossChainApplications/NativeTokenBridge/NativeTokenDestination"
+	nativetokensource "github.com/ava-labs/teleporter/abi-bindings/CrossChainApplications/NativeTokenBridge/NativeTokenSource"
+	deploymentUtils "github.com/ava-labs/teleporter/contract-deployment/utils"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+func NativeTokenBridge() {
+	const (
+		tokenReserve  = uint64(1e15)
+		valueToSend1  = tokenReserve / 4
+		valueToSend2  = tokenReserve
+		valueToReturn = valueToSend1 / 4
+
+		bridgeDeployerKeyStr               = "aad7440febfc8f9d73a58c3cb1f1754779a566978f9ebffcd4f4698e9b043985"
+		NativeTokenSourceByteCodeFile      = "./contracts/out/NativeTokenSource.sol/NativeTokenSource.json"
+		NativeTokenDestinationByteCodeFile = "./contracts/out/NativeTokenDestination.sol/NativeTokenDestination.json"
+	)
+	var (
+		ctx                       = context.Background()
+		nativeTokenBridgeDeployer = common.HexToAddress("0x1337cfd2dCff6270615B90938aCB1efE79801704")
+		tokenReceiverAddress      = common.HexToAddress("0x0123456789012345678901234567890123456789")
+	)
+
+	subnetA := utils.GetSubnetATestInfo()
+	subnetB := utils.GetSubnetBTestInfo()
+	teleporterContractAddress := utils.GetTeleporterContractAddress()
+
+	// Info we need to calculate for the test
+	nativeTokenBridgeDeployerPK, err := crypto.HexToECDSA(bridgeDeployerKeyStr)
+	Expect(err).Should(BeNil())
+	nativeTokenBridgeContractAddress, err := deploymentUtils.DeriveEVMContractAddress(nativeTokenBridgeDeployer, 0)
+	Expect(err).Should(BeNil())
+	log.Info("Native Token Bridge Contract Address: " + nativeTokenBridgeContractAddress.Hex())
+
+	{ // Deploy the contracts
+		nativeTokenSourceBytecode, err := deploymentUtils.ExtractByteCode(NativeTokenSourceByteCodeFile)
+		Expect(err).Should(BeNil())
+		chainATransactor, err := bind.NewKeyedTransactorWithChainID(nativeTokenBridgeDeployerPK, subnetA.ChainID)
+		Expect(err).Should(BeNil())
+		nativeTokenSourceAbi, err := nativetokensource.NativeTokenSourceMetaData.GetAbi()
+		Expect(err).Should(BeNil())
+		_, txA, _, err := bind.DeployContract(chainATransactor, *nativeTokenSourceAbi, nativeTokenSourceBytecode, subnetA.WSClient, teleporterContractAddress, subnetB.BlockchainID, nativeTokenBridgeContractAddress)
+		Expect(err).Should(BeNil())
```

Comment:
A `DeployContract` helper would be useful. We should include the calls to `WaitForTransaction` and `CodeAt` as well.